conu/backend/nspawn/image.py

```diff
@@ -0,0 +1,332 @@
+        return self.__class__(repository=source, tag=output_tag)
+
+    def tag_image(self, tag):
+        return self.create_snapshot(name=None, tag=tag)
+
+    def inspect(self, refresh=True):
+        return self.get_metadata(refresh=refresh)
+
+    def get_metadata(self, refresh=True):
+        if refresh or not self._metadata:
+            ident = self._id or self.get_full_name()
+            if not ident:
+                raise ConuException("This image does not have a valid identifier.")
+            self._metadata = output_decode(run_cmd(
+                ["machinectl", "--output=json", "show-image", ident], return_output=True))
+        return self._metadata
+
+    def rmi(self, force=False, via_name=False):
+        return run_cmd(["machinectl", "remove", self.get_id()])
+
+    def mount(self, mount_point=None):
+        return NspawnImageFS(self, mount_point=mount_point)
+
+    def _is_killed(self, name):
+        for foo in range(constants.DEFAULT_RETRYTIMEOUT):
+            time.sleep(constants.DEFAULT_SLEEP)
+            out = run_cmd(["machinectl", "status", name], ignore_status=True)
+            if out.exit_status != 0:
+                return True
+        raise ConuException("Unable to stop machine %s within %d" % (name, constants.DEFAULT_RETRYTIMEOUT))
+
+    def _is_booted(self, name, suffictinet_texts=["systemd-logind"]):
+        # optionally use: "Unit: machine"
+        for foo in range(constants.DEFAULT_RETRYTIMEOUT):
+            time.sleep(constants.DEFAULT_SLEEP)
+            out = run_cmd(["machinectl", "status", name], ignore_status=True, return_output=True)
+            for restr in suffictinet_texts:
+                if restr in out:
+                    time.sleep(constants.DEFAULT_SLEEP)
+                    return True
+        raise ConuException("Unable to start machine %s within %d" % (name, constants.DEFAULT_RETRYTIMEOUT))
+
+    def run(self, command=[], volumes=[], additional_opts=[], default_options=["-b"], foreground=False, name=None, *args, **kwargs):
+        logger.info("run container via binary in background")
+        machine_name = constants.CONU_ARTIFACT_TAG
+        if name:
+            machine_name += name
+        else:
+            machine_name += random_str()
+        if not kwargs:
+            kwargs = {}
+        if not foreground:
+            kwargs["stdout"] = subprocess.PIPE
+        additional_opts += default_options
+        if volumes:
+            additional_opts += self.get_volume_options(volumes=volumes)
+        logger.debug("starting NSPAWN")
+        systemd_command = ["systemd-nspawn", "--machine", machine_name, "-i", self.get_metadata()["Path"]] + additional_opts + command
+        logger.debug("Start command: %s" % " ".join(systemd_command))
+        self.container_process = subprocess.Popen(systemd_command, *args, **kwargs)
+        logger.info("machine: %s starting" % self.name)
+        if foreground:
+            self.container_process.wait()
```

Comment:
Also another topic for consideration:
 - in DockerBackend we just run the container, not dealing with result of the invocation - that is done later via methods is_running, wait_for_port, ... (whether is in run state or not). Would it make sense to unify what exactly is done in run method?